test/components/magic/adeptPowers/AdeptPowersUtilTest.js

```diff
@@ -0,0 +1,166 @@
+
+		expect(returnValue).to.eql({newSpell: adjustedSpell, isMystic});
+		expect(incrementAugmented).to.not.be.called;
+	});
+
+	it('Should return an 1 for levels if the power has levels', () => {
+		const isMystic = false;
+		const power = {
+			points: 1.5,
+			name: 'TestName',
+			levels: 'yes'
+		};
+		const pointsSpent = 3.5;
+		const maxPoints = 5;
+		const selectedOption = '';
+		const incrementAugmented = sinon.spy();
+		const adjustedSpell = Object.assign({}, power, {levels:1})
+
+		const returnValue = AdeptPowersUtil.generateAddActionDetails(isMystic, power, pointsSpent, maxPoints, selectedOption, incrementAugmented);
+
+		expect(returnValue).to.eql({newSpell: adjustedSpell, isMystic});
+		expect(incrementAugmented).to.not.be.called;
+	});
+
+	it('Should call incrementAugmented if there is enough points left to add', () => {
+		const isMystic = false;
+		const power = {
+			points: 1.5,
+			name: 'Improved Physical Attribute',
+			levels: 'yes',
+			bonus: {'selectattribute': {}}
+		};
+		const pointsSpent = 3.5;
+		const maxPoints = 5;
+		const selectedOption = '(Test)';
+		const incrementAugmented = sinon.spy();
+		const adjustedSpell = Object.assign({}, power, {levels: 1, name: power.name + selectedOption, bonus: selectedOption.replace(/[()]/g, '')})
+
+		const returnValue = AdeptPowersUtil.generateAddActionDetails(isMystic, power, pointsSpent, maxPoints, selectedOption, incrementAugmented);
+
+		expect(returnValue).to.eql({newSpell: adjustedSpell, isMystic});
+		expect(incrementAugmented).to.be.calledWith({attribute:selectedOption.replace(/[()]/g, '')});
+	});
+});
+
+describe('generateRemoveActionDetails', () => {
+	beforeEach(function() {
+		 sandbox.spy(AdeptPowersUtil, 'bonusDown');
+	});
+
+	afterEach(function() {
+		 sandbox.restore();
+	});
+
+	it('Should return the power index and isMystic value', () => {
+		const isMystic = false;
+		const power = {
+			bonus: 'N/A'
+		};
+		const index = 5;
+
+		const returnValue = AdeptPowersUtil.generateRemoveActionDetails(isMystic, power, index, () => {});
+
+		expect(returnValue).to.eql({powerIndex: index, isMystic});
+		expect(AdeptPowersUtil.bonusDown).to.not.be.called;
+	});
+
+	it('Should call decrementAugmented if it is improved attribute', () => {
+		const isMystic = true;
+		const baseName = 'Improved Physical Attribute';
+		const attribute = 'Test'
+		const power = {
+			name: baseName + '(' + attribute +')',
+			bonus: 'Test',
+			levels: 5
+		};
+		const index = 5;
+		const decrementAugmented = sinon.spy();
+
+		const returnValue = AdeptPowersUtil.generateRemoveActionDetails(isMystic, power, index, decrementAugmented);
+		expect(returnValue).to.eql({powerIndex: index, isMystic});
+		expect(decrementAugmented).to.be.calledWith({ attribute, decreaseBy: 5});
+	});
+});
+
+// Going to skip: raiseLevelAction, lowerLevelAction, and generatePowerDetailTablesRows not sure best way to test for now
```

Comment:
Because those functions are so complex, they'd be better candidates to be tested with an integration level test. But we don't have that level of testing just yet.